scrapy/utils/gz.py

```diff
@@ -18,17 +21,12 @@ def gunzip(data):
         except (IOError, EOFError, struct.error):
             # complete only if there is some data, otherwise re-raise
             # see issue 87 about catching struct.error
-            # some pages are quite small so output_list is empty and f.extrabuf
-            # contains the whole page content
-            if output_list or getattr(f, "extrabuf", None):
-                try:
-                    output_list.append(f.extrabuf[-f.extrasize :])
-                finally:
-                    break
-            else:
-                raise
+            # some pages are quite small so output_list is empty
+            if output_list:
+                break
+            raise
```

Comment:
Could you elaborate on this change?

Reply:
Right, I was going to do that but forgot.

`extrabuf` and `extrasize` were removed in 3.5 when the underlying logic was rewritten, they were never documented as far as I can see. Looks like this code was intended to salvage a partially decompressed chunk when possible, and now it's probably just not possible. The main purpose of this function though is to salvage previous completely decompressed chunks and it still does that, and tests in https://github.com/scrapy/scrapy/blob/master/tests/test_utils_gz.py only test that, as far as I can see.

Reply:
For the file in that test the first read1 call returns the data we expect. I'm not going to dive further into GzipFile implementations, it's possible that it's just more robust than before 3.5, I'm fine that the tests return the expected data with the code we have.